### PR TITLE
[FEATURE] create subscription - notification fields using ngsiv2

### PR DESCRIPTION
**Pull request by iariasleon**
@kzangeli @crbrox 
```
                    SUMMARY:
-------------------------------------------------
  - components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription.feature >> passed: 50, failed: 0, skipped: 0 and total: 50 with duration: 1.157 seconds.
  - components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature >> passed: 47, failed: 0, skipped: 61 and total: 108 with duration: 4.559 seconds.
  - components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_condition.feature >> passed: 97, failed: 0, skipped: 156 and total: 253 with duration: 5.615 seconds.
  - components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_entities.feature >> passed: 123, failed: 0, skipped: 47 and total: 170 with duration: 27.085 seconds.
-------------------------------------------------
  Date: Tuesday, May 03, 2016 14:47:26
-------------------------------------------------
4 features passed, 0 failed, 0 skipped
317 scenarios passed, 0 failed, 264 skipped
1826 steps passed, 0 failed, 1324 skipped, 0 undefined
Took 0m38.416s
```

A bug detected in GET types request and a test is marked  as bug

### CODE REVIEW

File: test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature [hunk 1]
```diff
@@ -225,29 +268,56 @@ Feature: create new subscriptions (POST) using NGSI v2. "POST" - /v2/subscriptio
       | http://www.foo.bar./       |
       | http://.www.foo.bar./      |
```
Comment (kzangeli) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature]: Why not add invalid ports as well?
E.g.:

    http://1.1.1.1:70000  # port is a short int, mas valid is lessa than 7000 (65535)
    http://1.1.1.1:abc    # port must be an integer

File: test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature [hunk 2]
```diff
@@ -394,13 +464,126 @@ Feature: create new subscriptions (POST) using NGSI v2. "POST" - /v2/subscriptio
...
+      | notification_attrs    | random=257              |
+      | expires               | 2016-04-05T14:00:00.00Z |
+    When create a new subscription
+    Then verify that receive a "Bad Request" http code
+    And verify an error response
+      | parameter   | value                                                              |
+      | error       | BadRequest                                                         |
+      | description | notification attribute name length: 257, max length supported: 256 |
 
+  @notification_attrs_two_equals @BUG_2101 @skip
```
Comment (kzangeli) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature]: I believe we decided this is the expected behavior of the broker.
If you ask to be notified with an attribute 'twice', the broker just does what you asked it to do.
The issue #2101 should be closed (check with @Fermin first).

File: test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_condition.feature [hunk 3]
```diff
@@ -537,49 +538,49 @@ Feature: create new subscriptions (POST) using NGSI v2. "POST" - /v2/subscriptio
...
-      | speed<99..190   |
-      | speed==99...190 |
-      | speed==99,,,190 |
+      | q                |
+      | speed>=69..90    |
+      | speed>69..90     |
+      | speed<=69..79    |
+      | speed<99..190    |
+      | speed==99....190 |
+      | speed==99,,,190  |
```
Comment (kzangeli) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_condition.feature]: Have you tested a 'negative' range?

  speed==100..1

Not sure how the broker reacts on a 'q' like that ...

[ Not sure how it should react:
* 1. Error,
* 2. Treat it like it is and never get a hit,
* 3. change it to 1..100 ]

File: test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature (hunk 2 above)
Comment (fgalan) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature]: I don't see any practical use case in which a reciver wants to receive two copies of the same attribute in the notificaiton. If we don't find any use case for that, then that should be fixed in a way or other (e.g. error report, "colapsing" duplicate notification.attrs attributes at subscription, time) and the #2101 (and this test case) make fully sense.

File: test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_condition.feature (hunk 3 above)
Comment (iariasleon) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_condition.feature]: Now, tested with:
```
      | q                |
      |------------------|
      | speed==100..1    |
      | speed==100..-1   |
      | speed==-100..-1  |
```
And all return: 201 Created

Comment (kzangeli) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_condition.feature]: Yeah, not **too** surprising, that is case 2. This subscription is useless as the 'q' will never evaluate to TRUE ...

The broker should probably detect this type of problem and return an error ...
@fgalan: your opinion?

File: test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature (hunk 1 above)
Comment (iariasleon) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature]: the first case is added, but the second, port is not an integer is covered in the line `http://localhost:dsfsdf`

File: test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature (hunk 2 above)
Comment (iariasleon) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_notification.feature]: NTC

File: test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_condition.feature (hunk 3 above)
Comment (fgalan) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_condition.feature]: Do you mean that the broker should be able of detecting "range inversion" and raise a syntax error? Ok for my, but as a low priority improvement. Please, create an issue about it (label "backlog") and response to this comment with its number to close this thread.

Comment (kzangeli) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_condition.feature]: Yes, that was my idea.  I see three options here, see above

Comment (kzangeli) [inline test/acceptance/behave/components/ngsiv2/subscriptions/create_a_new_subscription/create_a_new_subscription_subject_condition.feature]: Issue created: #2117

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**